src/net_processing.cpp

```diff
@@ -2437,6 +2437,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         if (!pfrom.IsInboundConn() && !pfrom.IsBlockOnlyConn()) {
```

Comment:
nit: is it worth replacing `if (... && !pfrom.IsBlockOnlyConn())` with `if (... && SetupAddrssRelay(pfrom))` since it already checks if the peer is block-relay-only?